Penteract/Assets/Scripts/AIMeleeGrunt.cpp

```diff
@@ -186,33 +194,44 @@ void AIMeleeGrunt::Update() {
 	case AIState::RUN:
 		movementScript->Seek(state, player->GetComponent<ComponentTransform>()->GetGlobalPosition(), speedToUse, true);
 		if (movementScript->CharacterInAttackRange(player, gruntCharacter.attackRange)) {
-			animation->SendTriggerSecondary("RunAttack");
-			attackRemaining = attackDuration;
-			if (agent) agent->SetMaxSpeed(0);
-			animation->SendTrigger("RunIdle");
+			int random = std::rand() % 100;
+			if (random < att1AbilityChance) { 
+				attackNumber = 1;
+				attackSpeed = att1AttackSpeed;
+				attackMovementSpeed = att1MovementSpeedWhileAttacking;
+			}
+			else if (random < att1AbilityChance + att2AbilityChance) {
+				attackNumber = 2;
+				attackSpeed = att2AttackSpeed;
+				attackMovementSpeed = att2MovementSpeedWhileAttacking;
+			}
+			else {
+				attackNumber = 3;
+				attackSpeed = att3AttackSpeed;
+				attackMovementSpeed = att3MovementSpeedWhileAttacking;
+			}
+
+			animation->SendTrigger("WalkForwardAttack" + std::to_string(attackNumber));
+			movementScript->SetClipSpeed(animation->GetCurrentState()->clipUid, attackSpeed);
 			if (audios[static_cast<int>(AudioType::ATTACK)]) audios[static_cast<int>(AudioType::ATTACK)]->Play();
 			state = AIState::ATTACK;
 		}
 		break;
 	case AIState::ATTACK:
-		attackRemaining -= Time::GetDeltaTime();
-		if (attackRemaining < 1.3f) { //frame 160
-			movementScript->Stop(); // stop seek
+		if (track) { 
+			movementScript->Orientate(player->GetComponent<ComponentTransform>()->GetGlobalPosition() - ownerTransform->GetGlobalPosition()); 
 		}
-		else {
-			movementScript->Orientate(player->GetComponent<ComponentTransform>()->GetGlobalPosition() - ownerTransform->GetGlobalPosition());
+		if (attackStep) {
+			movementScript->Seek(state, ownerTransform->GetGlobalPosition() + ownerTransform->GetFront()*10, attackMovementSpeed, false);
```

Comment:
Maybe expose the *10 so the distance multiplier can be tweaked?

Reply:
It is only to make sure u go front It would not have to change.